src/Features/Core/Portable/ImplementAbstractClass/ImplementAbstractClassData.cs

```diff
@@ -198,14 +235,98 @@ public async Task<Document> ImplementAbstractClassAsync(CancellationToken cancel
+                generator.MemberAccessExpression(throughExpression, @event.Name),
+                generator.IdentifierName("value")));
+
+            return CodeGenerationSymbolFactory.CreateAccessorSymbol(
+                attributes: default,
+                accessibility: Accessibility.NotApplicable,
+                statements: ImmutableArray.Create(statement));
         }
 
         private bool ShouldGenerateAccessor(IMethodSymbol? method)
             => method != null && this.ClassType.FindImplementationForAbstractMember(method) == null;
+
+        public IEnumerable<(ISymbol symbol, bool canDelegateAllMembers)> GetDelegatableMembers()
+        {
+            var fields = this.ClassType.GetMembers()
+                .OfType<IFieldSymbol>()
+                .Where(f => !f.IsImplicitlyDeclared)
+                .Where(f => InheritsFromOrEquals(f.Type, this.AbstractClassType))
+                .OfType<ISymbol>();
+
+            var properties = this.ClassType.GetMembers()
+                .OfType<IPropertySymbol>()
+                .Where(p => !p.IsImplicitlyDeclared && p.Parameters.Length == 0)
+                .Where(p => InheritsFromOrEquals(p.Type, this.AbstractClassType))
+                .OfType<ISymbol>();
+
+            // Have to make sure the field or prop has at least one unimplemented member exposed
+            // that we could actually call from our type.  For example, if we're calling through a
+            // type that isn't derived from us, then we can't access protected members.
+            foreach (var fieldOrProp in fields.Concat(properties))
+            {
+                var fieldOrPropType = fieldOrProp.GetMemberType();
+                var allUnimplementedMembers = _unimplementedMembers.SelectMany(t => t.members).ToImmutableArray();
+
+                var accessibleCount = allUnimplementedMembers.Count(m => m.IsAccessibleWithin(this.ClassType, throughType: fieldOrPropType));
+                if (accessibleCount > 0)
+                {
+                    // there was at least one unimplemented member that we could implement here
+                    // through one of our members.  Return this as a a delegatable member.  Also
+                    // indicate if we will be able to delegate all unimplemented members through
+                    // this.  If not, we'll let the user know that they can delegate through this
+                    // but that it will not fully fix the error.
+                    yield return (fieldOrProp, canDelegateAllMembers: accessibleCount == allUnimplementedMembers.Length);
+                }
+            }
+        }
+
+        private static bool InheritsFromOrEquals(ITypeSymbol type, ITypeSymbol baseType)
+            => GetBaseTypesAndThis(type).Contains(t => SymbolEquivalenceComparer.Instance.Equals(t, baseType));
+
+        private static IEnumerable<ITypeSymbol> GetBaseTypesAndThis(ITypeSymbol? type)
```

Comment:
Should this be an extension method with an optional parameter that enables looking at type parameter constraints, i.e. the if statement below?

Reply:
right now no one else needs it.  i'm not a fan of having as extensions in that case.